tests/integration/integration_test.go

```diff
@@ -1816,17 +1817,18 @@ func TestRunningViaCLIWrapper(t *testing.T) {
 		Cols: 80,
 	})
 	require.NoError(t, err)
-	defer ptmx.Close()
 
 	timeout := 3 * time.Minute
+	wait := 100 * time.Millisecond
 
 	go func() {
-		// Wait for the program to be ready before sending interrupt
-		for range int(timeout.Minutes()) {
+		// Wait for the program to be ready before sending the interrupt signal.
+		for range timeout.Milliseconds() / wait.Milliseconds() {
```

Comment:
Is this maybe easier to understand as 
```suggestion
		end := time.Now() + timeout
		for time.Now() < end {
```